File: Data Structures & Algorithms/buy-and-sell-crypto/submission-6.cpp
class Solution 
{
public:
    int maxProfit(vector<int>& prices) 
    {
        int res = 0;
        size_t left = 0, right = 0;
        while(right < prices.size())
        {
            if (prices[right] < prices[left])
            {
                left = right;
            }

            res = std::max((prices[right] - prices[left]), res);

            right++;
        }
        return res;   
    }
};

File: Data Structures & Algorithms/max-water-container/submission-6.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left, right = 0, len(heights) - 1
        max_area = 0

        while left < right:
            width = right - left
            cur_area = min(heights[left], heights[right]) * width
            max_area = max(cur_area, max_area)

            # Move the min bounds with hopes of finding a larger
            if heights[left] < heights[right]:
                left += 1
            else:
                right -= 1
        
        return max_area


